examples/r2i/classification.cc

```diff
@@ -0,0 +1,333 @@
+            << "-s 224 -b tensorflow -e libnormalize_inceptionv1.so -o libtop_sort_postprocessing.so -I input -O Softmax"
+            << std::endl;
+}
+
+std::unique_ptr<unsigned char[]> LoadImage(const std::string &path,
+    int reqwidth, int reqheight, int channels) {
+
+  int components_per_pixel;
+  int width;
+  int height;
+  const int scaled_size = channels * reqwidth * reqheight;
+  std::unique_ptr<unsigned char[]> scaled_ptr (new unsigned char[scaled_size]);
+  unsigned char *scaled = nullptr;
+
+  unsigned char *img = stbi_load(path.c_str(), &width, &height,
+                                 &components_per_pixel, channels);
+  if (!img) {
+    std::cerr << "The picture " << path << " could not be loaded" << std::endl;
+    return nullptr;
+  }
+
+  /* Scale image */
+  scaled = scaled_ptr.get();
+  stbir_resize_uint8(img, width, height, 0, scaled, reqwidth, reqheight, 0,
+                     channels);
+
+  std::cout << "  Image size: " << width << "x" << height << std::endl;
+  std::cout << "  Scaling to: " << reqwidth << "x" << reqheight << std::endl;
+  free (img);
+
+  return scaled_ptr;
+}
+
+bool ParseArgs (int &argc, char *argv[], std::string &image_path,
+                std::string &model_path, std::string &backend, int &size,
+                std::string &preprocess_path, std::string &postprocess_path,
+                std::string &in_node, std::string &out_node) {
+
+  int option = 0;
+  while ((option = getopt(argc, argv, "i:m:b:e:o:s:I:O:")) != -1) {
+    switch (option) {
+      case 'i' :
+        image_path = optarg;
+        break;
+      case 'm' :
+        model_path  = optarg;
+        break;
+      case 's' :
+        size = std::stoi (optarg);
+        break;
+      case 'b' :
+        backend = optarg;
+        break;
+      case 'e' :
+        preprocess_path = optarg;
+        break;
+      case 'o' :
+        postprocess_path = optarg;
+        break;
+      case 'I' :
+        in_node = optarg;
+        break;
+      case 'O' :
+        out_node = optarg;
+        break;
+      default:
+        return false;
+    }
+  }
+  return true;
+}
+
+std::string ToLowerCase(std::string string) {
+  std::string lowercase = string;
+
+  std::for_each(lowercase.begin(), lowercase.end(), [](char &c) {
+    c = std::tolower(c);
+  });
+  return lowercase;
+}
+
+int main (int argc, char *argv[]) {
+
+  r2i::RuntimeError error;
+  std::string model_path;
+  std::string image_path;
+  std::string backend;
+  std::string preprocess_path;
+  std::string postprocess_path;
+  std::string out_node;
+  std::string in_node;
+  r2i::FrameworkCode backend_code = r2i::MAX_FRAMEWORK;
+  int size = 0;
+
+  if (false == ParseArgs (argc, argv, image_path, model_path, backend,
+                          size, preprocess_path, postprocess_path,
+                          in_node, out_node)) {
+    PrintUsage ();
+    exit (EXIT_FAILURE);
+  }
+
+  if (image_path.empty() || model_path.empty () || preprocess_path.empty()
```

Comment:
Preprocess should not be mandatory. For example, TF1 Object Detection API Modules do no require one.

Reply:
I'm using the preprocess to get formats and sizes and dynamically process (scale) the input JPEG image. If the pre process is not specified, there is no way to get such information. There are 2 options:

1. Install options for the user to set width, height, format, channels
2. Ask the user (in this example) to always set a preprocess, it can be an empty preprocess, with the input format parameters specification only